### ISSUE TITLE
False-positive overlapping overload with TypeVar

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#2764]
**Describe the bug**
For the code below pyright reports that `Overload 2 will never be used` as it overlaps with overload 1. That seems to be a false-positive as the `reveal_type` examples show that although they overlap, both can still be used.

**To Reproduce**
Run pyright with `"typeCheckingMode": "strict"`.
```
error: Overload 2 for "func1" will never be used because its parameters overlap overload 1 (reportOverlappingOverload)
info: Type of "func1(C.a1)" is "list[Any]"
info: Type of "func1(C.a2)" is "list[int]"
```

**Expected behavior**
_No `reportOverlappingOverload` error_

**Screenshots or Code**
```python
from typing import overload, TypeVar, Any

T = TypeVar("T")

class A: ...

@overload
def func1(var: None) -> list[Any]:
    ...

@overload
def func1(var: T) -> list[T]:
    ...

def func1(var: T) -> list[T] | list[Any]:
    ...


class C:
    a1: None
    a2: int

reveal_type(func1(C.a1))
reveal_type(func1(C.a2))
```

**VS Code extension or command-line**
pyright 1.1.200

**Additional context**
Mypy does not report an error for it.
```
note: Revealed type is "builtins.list[Any]"
note: Revealed type is "builtins.list[builtins.int*]"
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.201, which I just published. It will also be included in the next release of pylance.
